spec.html

```diff
@@ -36251,12 +36251,20 @@ <h1>
             ): a non-negative integer
           </h1>
           <dl class="header">
+            <dt>description</dt>
+            <dd>It interprets _S_ as a sequence of UTF-16 encoded code points, as described in <emu-xref href="#sec-ecmascript-language-types-string-type"></emu-xref>, and returns the code unit index corresponding with code point index _e_, clipped to a maximum position just beyond the end of _S_.</dd>
           </dl>
           <emu-alg>
             1. If _S_ is the empty String, return 0.
-            1. Let _codepoints_ be StringToCodePoints(_S_).
```

Comment:
Well, StringToCodePoints does not return the code unit information that is needed in this algorithm. It could be derived from UTF16EncodeCodePoint or an analogous mathematical comparison of code unit value to 0xFFFF, but I find this use of [[CodeUnitCount]] information from CodePointAt to be more clear than alternatives like
```
1. Let _codepoints_ be StringToCodePoints(_S_).
1. Let _cpLen_ be the number of elements in _codepoints_.
1. Let _cuCount_ be 0.
1. Let _cpCount_ be 0.
1. Repeat, while _cpCount_ + 1 &lt; _cpIndex_,
  1. If _cpCount_ = _cpLen_, return the length of _S_.
  1. Let _cp_ be _codepoints_[_cpCount_].
  1. If _cp_ &le; 0xFFFF, set _cuCount_ to _cuCount_ + 1. Otherwise, set _cuCount_ to _cuCount_ + 2.
  1. Set _cpCount_ to _cpCount_ + 1.
1. Return _cuCount_.
```
or
```
1. Let _codepoints_ be StringToCodePoints(_S_).
1. Let _cpLen_ be the number of elements in _codepoints_.
1. Let _cpPrefix_ be a new empty List.
1. Let _cpCount_ be 0.
1. Repeat, while _cpCount_ + 1 &lt; _cpIndex_ and _cpCount_ &lt; _cpLen_,
  1. Append _codepoints_[_cpCount_] to _cpPrefix_.
  1. Set _cpCount_ to _cpCount_ + 1.
1. Let _strPrefix_ be ! CodePointsToString(_cpPrefix_).
1. Return the length of _strPrefix_.
```